### PR TITLE
Added line chart web component

### PR DESCRIPTION
**Pull request by dwnoble**
* Added line chart web component
* Changed `statVarDcid` to `variableDcid`
* Updated `draw` functions to accept a DOM element rather than ID because `document.getElementById` doesn't work when selecting in the shadow dom
* Changed web component mount elements from `span` to `div` - having them as `block` level elements seems more natural to me

Screenshot:

Screenshot 2023-06-23 at 12 13 46 PM

### CODE REVIEW

**Comment by dwnoble** [comment]
New updates:

* Renamed `placeDcid` attribute to `place`
* Renamed `variableDcid` attribute to `variable`

i still don't love `enclosedPlaceTypes` - maybe `childPlaceTypes` would be better? any opinions?

**Comment by shifucun** [comment]
> childPlaceTypes

I feel childPlaceTypes is better too.

**Review by chejennifer — COMMENTED** [review]
nice!

**Comment by pradh** [comment]
> > childPlaceTypes
> 
> I feel childPlaceTypes is better too.

+1.

I quickly looked over our other APIs.  It seems we largely call it just placeType, and in the description say "child" or "contained".  I like childPlaceType over placeType, since it is a bit more self-documenting.

File: static/js/chart/draw.ts [hunk 1]
```diff
@@ -927,7 +927,7 @@ function drawGroupBarChart(
     .attr("fill", (d) => colorFn(d.key));
 
   appendLegendElem(
-    id,
+    document.getElementById(id),
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply static/js/chart/draw.ts:930]: fixed

File: static/js/components/tiles/line_tile.tsx [hunk 2]
```diff
@@ -141,14 +141,13 @@ export const fetchData = async (props: LineTilePropType) => {
...
+  svgContainer: HTMLDivElement
 ): void {
-  const elem = document.getElementById(props.id);
   // TODO: Remove all cases of setting innerHTML directly.
-  elem.innerHTML = "";
+  svgContainer.innerHTML = "";
   const isCompleteLine = drawLineChart(
-    props.id,
-    props.svgChartWidth || elem.offsetWidth,
+    svgContainer,
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply static/js/components/tiles/line_tile.tsx:149]: Added. Previously i was getting the error:

```
Failed to get line tile result for: cat0-blk2-col0-tile0
```

But with that fix i'm no longer seeing the error

**Comment by dwnoble** [comment]
Renamed `enclosedPlaceType` to `childPlaceType`

**Review by juliawu — APPROVED** [review]

**Review by chejennifer — APPROVED** [review]
thanks for the updates!

**Pull request merged**